src/app/store/manage-tokens/manage-tokens.slice.ts

```diff
@@ -0,0 +1,45 @@
+import { useSelector } from 'react-redux';
+
+import { PayloadAction, createEntityAdapter, createSelector, createSlice } from '@reduxjs/toolkit';
+
+import type { RootState } from '..';
+
+interface TokenUserSetting {
+  id: string;
+  enabled: boolean;
+  accountIndex: number;
+}
+
+const manageTokensAdapter = createEntityAdapter<TokenUserSetting>();
+
+export const manageTokensSlice = createSlice({
+  name: 'manageTokens',
+  initialState: manageTokensAdapter.getInitialState(),
+  reducers: {
+    updateTokenEnableState(
+      state,
+      action: PayloadAction<{ id: string; enabled: boolean; accountIndex: number }>
+    ) {
+      manageTokensAdapter.setOne(state, {
+        id: action.payload.id,
+        enabled: action.payload.enabled,
+        accountIndex: action.payload.accountIndex,
+      });
+    },
```

Comment:
```suggestion
    updateTokenEnableState(
      state,
      action: PayloadAction<{ id: string; enabled: boolean; accountIndex: number }>
    ) {
      manageTokensAdapter.setOne(state, {
        id: action.payload.id,
        enabled: action.payload.enabled,
        accountIndex: action.payload.accountIndex,
      });
    },
```


This still kinda violates the "map to user actions, not state setters" practice. I'd go for something like `userTogglesTokenVisibility` etc